File: app/src/main/java/com/sofascoreacademy/sofascoreacademy/network/NetworkAPI.java
package com.sofascoreacademy.sofascoreacademy.network;

import com.sofascoreacademy.model.Team;
import com.sofascoreacademy.sofascoreacademy.model.Player;

import java.util.List;

import io.reactivex.rxjava3.core.Single;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface NetworkAPI {

    @GET("mobile/v4/mcc/{mcc}/teams")
    Single<List<Team>> mccTeams(@Path("mcc") int mcc);

    @GET("mobile/v4/mcc/{mcc}/players")
    Single<List<Player>> mccPlayers(@Path("mcc") int mcc);

    @GET("mobile/v4/team/{id}/details")
    Single<Team> teamDetails(@Path("id") int id);

    @GET("mobile/v4/player/{id}/details")
    Single<Player> playerDetails(@Path("id") int id);
}

File: app/src/main/java/com/sofascoreacademy/sofascoreacademy/network/NetworkAPI.kt
package com.sofascoreacademy.sofascoreacademy.network

import com.sofascoreacademy.model.Team
import com.sofascoreacademy.sofascoreacademy.model.Player
import retrofit2.http.GET
import retrofit2.http.Path

interface NetworkAPI {
    @GET("mobile/v4/mcc/{mcc}/teams")
    suspend fun mccTeams(@Path("mcc") mcc: Int): List<Team>

    @GET("mobile/v4/mcc/{mcc}/players")
    suspend fun mccPlayers(@Path("mcc") mcc: Int): List<Player>

    @GET("mobile/v4/team/{id}/details")
    suspend fun teamDetails(@Path("id") id: Int): Team

    @GET("mobile/v4/player/{id}/details")
    suspend fun playerDetails(@Path("id") id: Int): Player
}